txclib/utils.py

```diff
@@ -453,11 +453,37 @@ def color_text(text, color_name, bold=False):
         return text
 
 
-def files_in_project(curpath):
+def _can_walk(base, subdirectories):
     """
-    Iterate over the files in the project.
-    Return each file under ``curpath`` with its absolute name.
+    Determine if the first element of the 'subdirectories' list can be appended
+    to the 'base' path and still specify a directory.
     """
+    # Always consider the last part of the subdirectories the filename
+    if len(subdirectories) < 2:
+        return False
+    # Allow any directory name that doesn't contain the <lang> placeholder
+    return "<lang>" not in subdirectories[0]
+
+
+def get_project_files(curpath, expression):
+    """
+    Iterate over the files in the project that match the given expression.
+    Return a tuple with the absolute file path and the language code of the
+    language that is associated with it.
+    """
+    # Strip the reference to the current directory, if it exists
+    if expression.startswith("./"):
+        expression = expression[2:]
+    # Split the expression into parts
+    expression_parts = expression.split("/")
+    # Merge the expression's path into 'curpath' until the <lang> placeholder
+    # is specified in order to reduce the search tree
+    while _can_walk(curpath, expression_parts):
+        curpath = os.path.realpath(os.path.join(curpath, expression_parts[0]))
+        expression_parts = expression_parts[1:]
+    expr_re = regex_from_filefilter("/".join(expression_parts), curpath)
```

Comment:
@rigaspapas You shouldn't hardcode `/` here since it is not a valid separator for windows. Can't you do the same thing with `os.path.join`.

Reply:
Nice catch. I'll change it